File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if x > 2147483647 or x < -2147483648:
            return 0
        
        out = 0
        iterate = abs(x)
        while iterate != 0:
            out *= 10
            out += iterate % 10
            iterate = floor(iterate / 10)
        if x < 0:
            out *= -1;
        
        if out > 2147483647 or out < -2147483648:
            return 0
        
        return out
        

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        /*
        if (x < -2147483648 || x > 2147483647) {
        return 0;
    }*/
    long out = 0;
    long iterate = x;
    while (iterate != 0) {
        out *= 10;
        out += iterate % 10;
        iterate /= 10;
    }
    if (out < -2147483648 || out > 2147483647) {
        return 0;
    }
    return out;
    }
};